python/api/api_views.py

```diff
@@ -100,8 +100,9 @@ def get_api_url_kwargs(obj, request):
         if 'cluster' in request.path:
             kwargs['cluster_id'] = obj.cluster.id
     elif obj_type == 'component':
-        kwargs['service_id'] = obj.service.id
-        kwargs['cluster_id'] = obj.cluster.id
+        if 'cluster' in request.path:
```

Comment:
```suggestion
        elif obj_type == 'component' and 'cluster' in request.path:
```
@pisohaka 
Check plz sonarcube setup.
In report there is a code smell warning but check is green, can we fail it ?